public/js/gravityexport-lite.js

```diff
@@ -65,6 +65,30 @@ var gfexcel_sortable;
     };
 
     $(document).ready(function () {
-        $("#start_date, #end_date").datepicker({dateFormat: 'yy-mm-dd', changeMonth: true, changeYear: true});
+		$( '#start_date, #end_date' ).datepicker( { dateFormat: 'yy-mm-dd', changeMonth: true, changeYear: true } );
+
+		$( '#file_extension' ).on( 'change', function ( e ) {
+			const shortcode = $( '#embed_code' ).val();
+
+			$( '#embed_code' ).val( shortcode.replace( /type=\"[^\"]*\"/, `type="${ e.target.value }"` ) );
```

Comment:
The `type` might not be there. Needs to be added then too.

Reply:
@doekenorg, there is no need to do the extra check. Replacement will not take place if the `type` is not there.

Reply:
@mrcasual Yeah, but it should right? Because you want to update short code realtime. And if it isn't there, it didn't get added. Felt weird.

Reply:
@doekenorg, replacement does take place if `type` exists and doesn't if it doesn't exist :) IMHO, this just added unnecessary complexity: https://github.com/GravityKit/GravityExport-Lite/blob/93c5a2b101a1cd88c0e5f11302c488146fd91ee0/public/js/gravityexport-lite.js#L75-L82 If you insist, we can just do a simple `if (shortcode.includes('type="')) { return; }` before replacement.

Reply:
I'm not sure we are on the same page here. But from my pov: if we are going the route of updating the shortcode realtime (which is fine) then it should work regardless if you have a type set that point.

Because as soon as you change the type, it gets updated in the shortcode. But then you only see it after you saved the page. After that it gets replaced realtime. This solution unifies the solution, and makes for predictable behaviour.

But I'm not married to it. :-)

Reply:
Not sure. It needs to keep working with the default without the type. But It can very well be included by default. Makes the shortcode usage even more explicit which is a good thing, I guess.